lib/dal/src/property_editor/values.rs

```diff
@@ -213,11 +214,47 @@ impl PropertyEditorValues {
             child_values.insert(property_editor_value_id, child_property_editor_value_ids);
         }
 
-        Ok(PropertyEditorValues {
+        let mut property_editor_values = Self {
             root_value_id: root_property_editor_value_id,
             child_values,
             values,
-        })
+        };
+
+        // Before returning the resulting property editor values, we want to ensure we do not send
+        // up the encrypted secret key for values corresponding to secrets.
+        property_editor_values
+            .prepare_values_for_secrets_tree(ctx, component_id)
+            .await?;
+
+        Ok(property_editor_values)
+    }
+
+    async fn prepare_values_for_secrets_tree(
+        &mut self,
+        ctx: &DalContext,
+        component_id: ComponentId,
+    ) -> PropertyEditorResult<()> {
+        let schema_variant_id = Component::schema_variant_id(ctx, component_id).await?;
+
+        let secret_object_prop_id =
+            Prop::find_prop_id_by_path(ctx, schema_variant_id, &PropPath::new(["root", "secrets"]))
+                .await?;
+
+        for secret_prop_id in Prop::direct_child_prop_ids(ctx, secret_object_prop_id).await? {
+            let attribute_value_id = self.find_by_prop_id(secret_prop_id).ok_or(
+                PropertyEditorError::PropertyEditorValueNotFoundByPropId(secret_object_prop_id),
+            )?;
+
+            if let Some(property_editor_value) = self.values.get_mut(&attribute_value_id.into()) {
+                if Value::Null != property_editor_value.value {
+                    property_editor_value.value = Secret::payload_for_property_editor_values(
+                        property_editor_value.value.clone(),
+                    )?;
+                }
+            }
```

Comment:
Remove the key from the values here. This has negligible performance impact in the real world from looking at round trip times from live routes.